RyleyGG/apluschiever
# Initial infra

# Issue
Closes #2, #3, #4, #5, #13

# Description
This PR implements the basis of the project and its infrastructure. This includes a docker container setup for the API and postgres database, a basic FastAPI layout with implemented authentication endpoints, and a pytest test suite. The authentication is implemented following the OAuth2 framework.

# Type Of Change

# Test Steps
* Follow the readme instructions and ensure the project can be stood up without any issues.
* Run the pytest test suite and ensure all test cases pass as expected.

# Checklist
